hydra-node/src/Hydra/Chain/Direct/Handlers.hs

```diff
@@ -230,21 +231,19 @@ chainSyncHandler tracer callback getTimeHandle ctx =
                       }
                 }
 
--- | Hardcoded grace time for close transaction to be valid.
--- TODO: make it a node configuration parameter
-closeGraceTime :: SlotNo
-closeGraceTime = 100
-
-fromPostChainTx ::
+prepareTxToPost ::
   (MonadSTM m, MonadThrow (STM m)) =>
   TimeHandle ->
   TinyWallet m ->
   ChainContext ->
   ChainStateType Tx ->
   PostChainTx Tx ->
   STM m Tx
-fromPostChainTx timeHandle wallet ctx cst@ChainStateAt{chainState} tx = do
-  pointInTime <- throwLeft currentPointInTime
+prepareTxToPost timeHandle wallet ctx cst@ChainStateAt{chainState} tx = do
+  pointInTime@(currentSlot, _) <- throwLeft currentPointInTime
+  let (UnsafeContestationPeriod cpNatural) = contestationPeriod ctx
+      -- we are using the contestationPeriod as tx close grace time. See ADR21 for context
+      closeGraceTime = SlotNo $ fromIntegral cpNatural
```

Comment:
Ok this is also puzzling a bit. Previously we had `closeGraceTime` expressed in slots and hardcoded to 100 slots. That was fed into the `adjustPointInTime` for close/contest tx. Right now we have CP as number of seconds and we want to use that instead of `closeGraceTime`. I don't see a way of converting number of seconds to slots immediately. I tried adding CP to current slot and use that for tx validity but the tests are not happy that way. I'll try adding CP value to currentTime and then convert it to slots to see if that helps.

Reply:
Ok, that is just wrong. What we need here is to actually _divide_ cp seconds to the slot length for the underlying network in order to get what these seconds mean in slots but that doesn't seem possible with the current code. Maybe extend the `TimeHandle` ?

Reply:
Another simpler option - subtract the calculated end slot and current slot to get the slot difference.

Reply:
Ok this approach worked. I'll stop rambling.